pkg/controller/backup_repository_controller.go

```diff
@@ -194,8 +204,14 @@ func (r *BackupRepoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	bsl, bslErr := r.getBSL(ctx, backupRepo)
+	if bslErr != nil {
+		log.WithError(bslErr).Error("Fail to update BackupRepository status, when BSL is not found for that BackupRepository.")
```

Comment:
The controller does more things than just updating BackupRepository status, so we can make the message more clear to indicate that nothing could be done on the repo since the BSL is not found.

Reply:
Modified.